app/src/main/java/org/stepik/android/adaptive/pdd/ui/adapter/RatingAdapter.kt

```diff
@@ -0,0 +1,85 @@
+package org.stepik.android.adaptive.pdd.ui.adapter
+
+import android.support.v4.content.ContextCompat
+import android.support.v4.graphics.drawable.DrawableCompat
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import org.stepik.android.adaptive.pdd.R
+import org.stepik.android.adaptive.pdd.data.SharedPreferenceMgr
+import org.stepik.android.adaptive.pdd.data.model.RatingItem
+import org.stepik.android.adaptive.pdd.databinding.ItemRatingBinding
+
+class RatingAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+    companion object {
+        private val RATING_ITEM_VIEW_TYPE = 1
+        private val SEPARATOR_VIEW_TYPE = 2
+
+        private val SEPARATOR = -1L
+    }
+
+    private val items = ArrayList<RatingItem>()
+
+    override fun getItemViewType(position: Int) =
+        if (items[position].user != SEPARATOR) {
+            RATING_ITEM_VIEW_TYPE
+        } else {
+            SEPARATOR_VIEW_TYPE
+        }
+
+    override fun getItemCount() = items.size
+
+    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int) =
+            if (viewType == RATING_ITEM_VIEW_TYPE) {
```

Comment:
Use switch by viewType and in else statement throw `IllegalStateException`. This approach helps to avoid bugs in the future